### PR TITLE
Update es signing requests using new round tripper method from dp-net

### PR DESCRIPTION
**Pull request by nshumoogum**
### What

Dependency upgrade of dp-net version 2 to use new method to create http transport layer for signing requests to elasticsearch, using `NewClientWithTransport`.

### How to review

Check changes make sense, see documentation of how to use awsauth package [here](https://github.com/ONSdigital/dp-net/tree/feature/restructure-aws-signing-round-tripper/awsauth#awsauth-package)

This can only be tested in an aws account, so will need to deploy to staging environment to see it working.

### Who can review

Anyone

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,15 +43,40 @@ environment variables, or with a link to a configuration guide.
...
-| SIGN_ELASTICSEARCH_REQUESTS | false                   | Boolean flag to identify whether elasticsearch requests via elastic API need to be signed if elasticsearch cluster is running in aws
-| HEALTHCHECK_CRITICAL_TIMEOUT| 90s                     | Time to wait until an unhealthy dependent propagates its state to make this app unhealthy (`time.Duration` format)
-| HEALTHCHECK_INTERVAL        | 30s                     | Time between self-healthchecks (`time.Duration` format)
-| ZEBEDEE_URL                 | http://localhost:8082   | The URL to Zebedee (for authorisation)
+| AWS_FILENAME                 | ""                       | The AWS file location for finding credentials to sign AWS http requests
+| AWS_PROFILE                  | ""                       | The AWS profile to use from credentials file to sign AWS http requests
+| AWS_REGION                   | eu-west-1                | The AWS region to use when signing requests with AWS SDK
+| AWS_SERVICE                  | "es"                     | The aws service that the AWS SDK signing mechanism needs to sign a request
```
Comment (justinpjose) [inline README.md:49]:
```suggestion
| AWS_SERVICE                  | "es"                     | The AWS service that the AWS SDK signing mechanism needs to sign a request
```

File: config/config_test.go [hunk 2]
```diff
@@ -17,8 +17,8 @@ func TestSpec(t *testing.T) {
 				So(err, ShouldBeNil)
 			})
 			Convey("The values should be set to the expected defaults", func() {
-				So(cfg.AwsRegion, ShouldEqual, "eu-west-1")
-				So(cfg.AwsService, ShouldEqual, "es")
+				So(cfg.Aws.Region, ShouldEqual, "eu-west-1")
```
Comment (justinpjose) [inline config/config_test.go:20]: Adding a test for `Aws.Filename`, `Aws.Profile` and `Aws.TLSInsecureSkipVerify`
But most importantly `Aws.TLSInsecureSkipVerify` as it should never be true but only for development. That little extra

**Review by justinpjose — COMMENTED** [review]
Outside of this, can you update the license information in the `README.md` and `LICENCE.md `to `2022` and also, the go version in the `ci` directory to `1.17.7`

File: config/config.go [hunk 3]
```diff
@@ -9,8 +9,7 @@ import (
 
 // Config is the search API handler config
 type Config struct {
-	AwsRegion                  string        `envconfig:"AWS_REGION"`
-	AwsService                 string        `envconfig:"AWS_SERVICE"`
+	Aws                        AWS
```
Comment (rahulPalliyalil) [inline config/config.go:12]: (suggestion): would be better to rename this to ```AWS``` for example AWSRegion, AWSService etc.

**Review by rahulPalliyalil — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -25,6 +25,10 @@ lint:
 	go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.43.0
 	golangci-lint run ./...
 
+.PHONY: local
+local:
+	export ELASTIC_SEARCH_URL=https://localhost:9200; export AWS_TLS_INSECURE_SKIP_VERIFY=true; export SIGN_ELASTICSEARCH_REQUESTS=true; export AWS_PROFILE=development; export AWS_FILENAME=$(HOME)/.aws/credentials; HUMAN_LOG=1 go run $(LDFLAGS) -race main.go
```
[the remark this answers is not in the document]
Comment (nshumoogum) [reply Makefile:30]: Each separate (indented) line is run in its own shell, so doesn't carry-forward. I will extend your example so it does carry forward using `\`

**Review by rahulPalliyalil — APPROVED** [review]

**Pull request merged**